src/jdk.compiler/share/classes/com/sun/tools/javac/code/Types.java

```diff
@@ -1771,6 +1756,14 @@ public Boolean visitClassType(ClassType t, Type s) {
                 }
 
                 if (s.hasTag(CLASS) || s.hasTag(ARRAY)) {
+                    if (isValue(t)) {
+                        // (s) Value ? == (s) Value.ref
```

Comment:
I didn't get these comments

Reply:
It is saying the predicate is a value type is castable to a type s is the same as the predicate is its reference projection castable to s.